### ISSUE TITLE
IDE0079 should not warn when suppressing CA1716

### ISSUE THREAD

**Issue by JohanLarsson** [dotnet/roslyn#50203]
```cs
#pragma warning disable CA1716 // Identifiers should not match keywords
namespace Select
#pragma warning restore CA1716 // Identifiers should not match keywords
{
}
```

![Animation](https://user-images.githubusercontent.com/1640096/103446308-70621b00-4c7e-11eb-9388-64e0e8ccb728.gif)

**Comment by sharwell** [comment]
This is likely caused by non-deterministic reporting here:
https://github.com/dotnet/roslyn-analyzers/blob/3fb878177fb161344a57784957485450544e0f84/src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/IdentifiersShouldNotMatchKeywords.cs#L152-L156

**Comment by Youssef1313** [comment]
@sharwell Should this be moved to roslyn-analyzers repo then?

**Comment by mavasani** [comment]
The issue here seems to be that CA1716 has multiple descriptors with same ID, but only one of them reported at CompilationEnd, rest are reported during Symbol callback. The diagnostic reported from CompilationEnd is not reported in live analysis, hence it causes IDE0079 false positive. IDE0079 has logic to skip CompilationEnd diagnostics completely, so I'll check why that is not working as expected.

**Comment by mavasani** [comment]
Actually, my analysis above is wrong (I was looking at the wrong analyzer). @sharwell's analysis is more closer to the underlying issue, and might even be something that @sharwell and I need to investigate more closely.

- CA1716 analyzer reports the diagnostic on namespace symbol at https://github.com/dotnet/roslyn-analyzers/blob/3fb878177fb161344a57784957485450544e0f84/src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/IdentifiersShouldNotMatchKeywords.cs#L161
- This seems to lead to operation canceled exception on majority of occasions on first callback for the type - this seems to be due to the fact that the analyzer is registering NamedType symbol callbacks but then reporting diagnostic on containing namespace location, and the IsGeneratedCode check on that seems to cause this.
- Analyzer driver makes another callback for the same symbol on this compilation, as cancellation was throw during the symbol callback itself. I would have assumed that cancelation firing would lead us to drop the compilationWithAnalyzers instance completely for analysis, but it seems not to be the case. I'll investigate why this is happening in the IDE.
